### PR TITLE
Update pending intent flags

### PR DESCRIPTION
**Pull request by ionutbedregeanu**
For #6244
Since Android 12 we need to set PendingIntent mutability explicitly
This additional requirement improves your app's security.
All pending intents are updated to contain also the flag FLAG_IMMUTABLE when the version of
Android SDK is bigger than 30
FLAG_IMMUTABLE -> Flag indicating that the created PendingIntent should be immutable.

#### Pixel3a Api 29

https://user-images.githubusercontent.com/11731652/150143492-ff23a901-6822-43c5-b312-ed755514dac3.mp4

#### Pixel3a Api 31

https://user-images.githubusercontent.com/11731652/150143543-1e923d14-537c-437c-b967-17985f29aa7d.mp4

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### CODE REVIEW

**Comment by ionutbedregeanu** [comment]
Marked as draft because the app is still crashing

File: app/src/main/java/org/mozilla/focus/session/SessionNotificationService.kt [hunk 1]
```diff
@@ -102,27 +102,47 @@ class SessionNotificationService : Service() {
     }
 
     private fun createNotificationIntent(): PendingIntent {
+        val notificationIntentFlags = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```
Comment (ionutbedregeanu) [inline app/src/main/java/org/mozilla/focus/session/SessionNotificationService.kt:105]: I will also add comments in code to explain why

File: app/src/androidTest/java/org/mozilla/focus/activity/CustomTabTest.kt [hunk 2]
```diff
@@ -124,10 +125,15 @@ class CustomTabTest {
 
     @Suppress("Deprecation")
     private fun createCustomTabIntent(pageUrl: String): Intent {
+        val pendingIntentFlags = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```
Comment (Mugurell) [inline app/src/androidTest/java/org/mozilla/focus/activity/CustomTabTest.kt:128]: nit: We could have this as maybe an extension to encapsulate this behavior and reduce the amount of duplicated code and documentation.
Something like
```
val PendingIntent.defaultFlags
    get() = if (Build.VERSION.SDK_INT > Build.VERSION_CODES.M) {
        PendingIntent.FLAG_IMMUTABLE
    } else {
        0 // No flags. Default behavior.
    }
```

**Comment by jonalmeida** [comment]
Tried out this patch with https://github.com/mozilla-mobile/android-components/pull/11578 and it runs fine,

**Review by Mugurell — APPROVED** [review]
Thanks @ionutbedregeanu and @jonalmeida for all the testing and work here!

Awesome to also see tests though it seems like we need to standardize an approach for running tests under different `SDK_INT` and probably should be part of a separate ticket.

**Pull request merged**